Scripts/Services/Harvest/Mining.cs

```diff
@@ -221,6 +205,14 @@ public override Type GetResourceType(Mobile from, Item tool, HarvestDefinition d
 
                 PlayerMobile pm = from as PlayerMobile;
 
+                if (tool is ImprovedRockHammer)
```

Comment:
Can we just do this:
double chance = tool is RockHammer ? 0.50 : tool is ImprovedRockHammer ? 1.0 : 0.15;

Or have ImprovedRockHammer derive from rock hammer and make an overridable property to define bonus chance. That way if there are more introduced, its easier to handled.